numpy/_core/strings.pyi

```diff
@@ -1,70 +1,103 @@
-from typing import Any, overload
+from typing import Any, overload, TypeAlias
 
 import numpy as np
 from numpy._typing import (
     NDArray,
     _ArrayLikeStr_co as U_co,
     _ArrayLikeBytes_co as S_co,
     _ArrayLikeInt_co as i_co,
-    _ArrayLikeBool_co as b_co,
+    _ArrayLikeString_co as T_co,
+    _ArrayLikeAnyString_co as UST_co,
+    _Shape,
+    _SupportsArray,
 )
 
+
+AR_T: TypeAlias = np.ndarray[_Shape, np.dtypes.StringDType]
```

Comment:
It's probably cleaner to rename this using CamelCase convention and add a leading underscore